third_party/cabal2bazel/bzl/cabal_paths.bzl

```diff
@@ -31,19 +31,33 @@ def _impl_path_module_gen(ctx):
   base_dir = ctx.label.package + (
       ("/" + ctx.attr.data_dir) if ctx.attr.data_dir else "")
 
-  ctx.template_action(
+  # If workspace_root is not empty, we have an external repo.
+  # We need to relativize the paths.
+  if ctx.label.workspace_root != "":
+    ctx.template_action(
       template=ctx.file._template,
       output=paths_file,
       substitutions={
-          "%{module}": ctx.attr.module,
-          "%{base_dir}": paths.join(
-              # TODO: this probably won't work for packages not in external
-              # repositories.  See:
-              # https://github.com/bazelbuild/bazel/wiki/Updating-the-runfiles-tree-structure
-              "..", paths.relativize(ctx.label.workspace_root, "external"), base_dir),
-          "%{version}": str(ctx.attr.version),
+        "%{module}": ctx.attr.module,
+        "%{base_dir}": paths.join(
+          # TODO: this probably won't work for packages not in external
```

Comment:
Have your changes affected the context of this TODO?

Reply:
Yeah that shouldn't be here.